docs(users): authentication with auth-oidc plugin

thomas-dkmt opened the pull request:
### Description

The documentation now explains how to set up authentication in Xen Orchestra using the OpenID Connect plugin. 

It includes both admin-side configuration and user-side login steps, and should address this issue: https://github.com/vatesfr/xen-orchestra/issues/8225

### Checklist

- Commit
  - Title follows [commit conventions](https://bit.ly/commit-conventions)
  - Reference the relevant issue (`Fixes #007`, `See xoa-support#42`, `See https://...`)
  - If bug fix, add `Introduced by`
- Changelog
  - If visible by XOA users, add changelog entry
  - Update "Packages to release" in `CHANGELOG.unreleased.md`
- PR
  - If UI changes, add screenshots
  - If not finished or not tested, open as _Draft_

### Review process

> This 2-passes review process aims to:
>
> - develop skills of junior reviewers
> - limit the workload for senior reviewers
> - limit the number of unnecessary changes by the _author_

1. The _author_ creates a PR.
2. Review process:
   1. The _author_ assigns the _junior reviewer_.
   2. The _junior reviewer_ conducts their review:
      - Resolves their comments if they are addressed.
      - Adds comments if necessary or approves the PR.
   3. The _junior reviewer_ assigns the _senior reviewer_.
   4. The _senior reviewer_ conducts their review:
      - If there are no unresolved comments on the PR → merge.
      - Otherwise, we continue with **3.**
3. The _author_ responds to comments and/or makes corrections, and we go back to **2.**

Notes:

1. The _author_ can request a review at any time, even if the PR is still a _Draft_.
2. In theory, there should not be more than one reviewer at a time.
3. The _author_ should not make any changes:
   - When a reviewer is assigned.
   - Between the _junior_ and _senior_ reviews.

thomas-dkmt committed:
doc(xoa): Add documentation for authentication with auth-oidc plugin

The documentation now explains how to set up authentication in
Xen Orchestra using the OpenID Connect plugin. It includes both
admin-side configuration and user-side login steps.

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

users.md: Simplify label description

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

nathanael-h commented on docs/docs/users.md:
```diff
@@ -95,6 +95,71 @@ Importing the groups doesn't import their members. The users will still be impor
+   - **Issuer**: The expected issuer string from the identity provider.
+   - **Token URL**: The URL for retrieving access tokens.
+   - **User info URL**: The URL for user profile information.
+   - **Username field**: Field to use as the Xen Orchestra username (e.g., `displayName`, `username`, or `email`).
```
We should not suggest `displayName` as it is common that the end user can modify this value themself, and by doing this they could then impersonnate someoneelse if the OIDC Auth plugin use this value as the XO username. Usually the value we should match against is the [subject ID](https://openid.net/specs/openid-connect-core-1_0.html#SubjectIDTypes), `sub`. 
```suggestion
   - **Username field**: Field to use as the Xen Orchestra username (e.g., `sub`, `username`, or `email`).
```

nathanael-h commented:
Thanks @thomas-dkmt !

julien-f commented on docs/docs/users.md:
```diff
@@ -95,6 +95,81 @@ Importing the groups doesn't import their members. The users will still be impor
+
+#### Overview
+
+The OpenID Connect (OIDC) plugin (`auth-oidc`) allows Xen Orchestra to integrate with identity providers that support the OIDC protocol.
```
```suggestion
The `auth-oidc` plugin allows Xen Orchestra to integrate with identity providers that support the [OpenID Connect](https://en.wikipedia.org/wiki/OpenID#OpenID_Connect_(OIDC)) protocol.
```

julien-f commented on docs/docs/users.md:
```diff
@@ -95,6 +95,75 @@ Importing the groups doesn't import their members. The users will still be impor
+3. Enter your credentials:
+   - **Username or email**: Your account's username or email address.
+   - **Password**: Your account's password.
+4. Click **Sign In**.
```
AFAIK this is really dependent on the identity provider, not sure we should make assumptions.

thomas-dkmt replied:
Fixed by https://github.com/vatesfr/xen-orchestra/pull/8230/commits/ab2d830ae61e7b60cf784f0d556c57406e9d7545

julien-f commented on docs/docs/users.md:
```diff
@@ -95,6 +95,75 @@ Importing the groups doesn't import their members. The users will still be impor
+   - **Token URL**: The URL for retrieving access tokens.
+   - **User info URL**: The URL for user profile information.
+   - **Username field**: Field to use as the Xen Orchestra username (e.g., `sub`, `username`, or `email`).
+   - **Scopes**: List of OIDC scopes for profile information, separated by a single space. Note: The `openid` scope is included automatically.
```
Wouldn't it make sense to add the settings' description directly in the plugin's UI instead?

thomas-dkmt replied:
Why not, but in the meantime, should we keep it in the documentation?

julien-f replied:
I don't think so, it's trivial to add description to fields: https://github.com/vatesfr/xen-orchestra/blob/4296ca7e9b8faf1f63338f125b6a6a7108a23d98/packages/xo-server-auth-oidc/index.js#L14

thomas-dkmt committed:
users.md: Remove description of username/email/password

How these fields are used is dependent on the identity provider,
so we shouldn't make assumptions.
Therefore, we're removing the description

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

thomas-dkmt committed:
users.md: Remove unnecessary field descriptions

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

julien-f commented on docs/docs/users.md:
```diff
@@ -95,6 +95,71 @@ Importing the groups doesn't import their members. The users will still be impor
+   - **Token URL**
+   - **User info URL**
+   - **Username field**
+   - **Scopes**
```
I don't think this is necessary, we only need to say to fill the settings before saving and activating.

julien-f committed:
add settings description in the plugin

julien-f committed:
update image

julien-f committed:
optimize image

thomas-dkmt committed:
users.md: Remove unnecessary field list

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

thomas-dkmt committed:
users.md: Remove mandatory field details

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

thomas-dkmt committed:
users.md: Add mention of the optional URL field in the base configuration

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

julien-f committed:
format with Prettier

julien-f committed:
fix Markdown

julien-f committed:
Merge branch 'master' into doc-auth-oidc-plugin
